bibtex-completion.el

```diff
@@ -284,13 +284,29 @@ In the format string, expressions like \"${author:36}\", \"${title:*}\", etc, ar
   "A cache storing the hash of the bibliography content and the corresponding list of entries, for each bibliography file, obtained when the bibliography was last parsed. When the current bibliography hash is identical to the cached hash, the cached list of candidates is reused, otherwise the bibliography file is reparsed.")
 
 
+(defun bibtex-completion-normalize-bibliography (&optional type)
+  "Returns a list of bibliography file(s) in `bibtex-completion-bibliography'. If there are org-mode bibliography-files, their corresponding bibtex files are listed as well, unless TYPE is 'main. If TYPE is 'bibtex, org-mode bibliography-files are instead replaced with their associated bibtex files."
+  (delete-dups
+   (cl-loop
+    for bib-file in (-flatten (list bibtex-completion-bibliography))
+    for main-file = (if (consp bib-file)
+                        (car bib-file)
+                      bib-file)
+    for bibtex-file = (if (consp bib-file)
+                          (cdr bib-file)
+                        (concat (file-name-sans-extension main-file) ".bib"))
+    unless (equal type 'bibtex)
```

Comment:
There are three different "normalized" bibliographies we may need:
- only the "main" files (whether .bib or .org), but not the .bib file converted from a .org one. We need this for `bibtex-completion-show-entry`.
- only the bibtex files (so for a .org file, we only want the converted .bib file). We need this for `bibtex-completion-candidates` and `bibtex-completion-get-entry1`.
- all files (so for a .org file we want both this file and its converted .bib file). We need this in `bibtex-completion-init`.

So I chose to assign the types `'main` and `'bibtex` to the first and second case, respectively, and any other value (`nil` in particular) for the third one. Hence line 298 applies whenever it is not `'bibtex`, not only when it is `'main`. Similarly, line 300 applies whenever it is not `'main`, not only when it is `'bibtex`.

But it is surely possible to reformulate this. For instance, we could explicitly assign the type `'full` to the thrid case (and then make the type argument required rather than optional). Then in line 298: `(when (member type '(main full)) ...` and in line 300 `(when (member type '(bibtex full)) ...`. Just let me know how you would like to rephrase things and I will make the change.